composer/trainer/trainer.py

```diff
@@ -887,6 +895,28 @@ def __init__(
             precision = Precision(precision)
         _validate_precision(precision, device)
 
+        # check if provided model is compiled or not
+        is_torch_2_0 = using_torch_2_0()
+        is_model_compiled = False
+        if is_torch_2_0:
+            from torch._dynamo import OptimizedModule
+            if isinstance(model, OptimizedModule):
```

Comment:
do we need to recursively check all module for being `OptimizedModule` like we do for FSDP?

Reply:
I don't think so since `torch.compile` returns a `torch._dynamo.OptimizedModule` which wraps the ComposerModel passed in as an argument but does not actually inherit from it. So just checking the first layer should suffice.